### ISSUE TITLE
[BUG] [REGRESSION] crash (and loop crash) when using keycloakoidc authconfig

### ISSUE THREAD

**Issue by jaredkipe** [rancher/rancher#40631]
Rancher 2.7.0

Setup KeycloakOIDC provider just like what is/has been working in 2.6.5 for months. Users can login, but out of the blue user logins will crash Rancher.  

Rancher can be recovered by deleting the tokens (k8s objects in the rancher cluster) (at least initially, as them logging back in will cause it again).  OR by disabling the keycloakoidc authconfig (k8s object in the rancher cluster).

**Worse than that.  It is a regression as I have the same crash after upgrading 2.6.5 to 2.6.10**

```
│ 2023/02/06 18:49:47 [INFO] Watching metadata for cluster.x-k8s.io/v1alpha3, Kind=MachineHealthCheck                                                                      │
│ 2023/02/06 18:49:47 [INFO] Starting cluster.x-k8s.io/v1alpha3, Kind=MachineDeployment controller                                                                         │
│ 2023/02/06 18:49:47 [INFO] [clean-catalog-orphan-bindings] Processing 4 rolebindings                                                                                     │
│ 2023/02/06 18:49:47 [INFO] [clean-catalog-orphan-bindings] Deleting orphaned role global-catalog                                                                         │
│ 2023/02/06 18:49:47 [WARNING] [clean-catalog-orphan-bindings] Error when deleting role global-catalog, roles.rbac.authorization.k8s.io "global-catalog" not found        │
│ 2023/02/06 18:49:47 [WARNING] [CleanupOrphanCatalogBindingsDone] error during orphan binding cleanup: roles.rbac.authorization.k8s.io "global-catalog" not found         │
│ 2023/02/06 18:49:47 [ERROR] failed to cleanup orphan catalog bindings                                                                                                    │
│ 2023/02/06 18:49:47 [INFO] Watching metadata for cluster.x-k8s.io/v1alpha3, Kind=MachineSet                                                                              │
│ 2023/02/06 18:49:47 [INFO] Starting cluster.x-k8s.io/v1alpha3, Kind=MachineHealthCheck controller                                                                        │
│ E0206 18:49:47.284742      33 runtime.go:79] Observed a panic: runtime.boundsError{x:1, y:1, signed:true, code:0x0} (runtime error: index out of range [1] with length 1 │
│ goroutine 3812 [running]:                                                                                                                                                │
│ k8s.io/apimachinery/pkg/util/runtime.logPanic({0x3c4d0a0?, 0xc006148f30})                                                                                                │
│     /go/pkg/mod/k8s.io/apimachinery@v0.24.5/pkg/util/runtime/runtime.go:75 +0x99                                                                                         │
│ k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc007889b50?})                                                                                              │
│     /go/pkg/mod/k8s.io/apimachinery@v0.24.5/pkg/util/runtime/runtime.go:49 +0x75                                                                                         │
│ panic({0x3c4d0a0, 0xc006148f30})                                                                                                                                         │
│     /usr/lib64/go/1.19/src/runtime/panic.go:884 +0x212                                                                                                                   │
│ github.com/rancher/rancher/pkg/auth/providers/keycloakoidc.getSearchURL({0xc012de5980?, 0x0?})                                                                           │
│     /go/src/github.com/rancher/rancher/pkg/auth/providers/keycloakoidc/keycloak_client.go:172 +0xe5                                                                      │
│ github.com/rancher/rancher/pkg/auth/providers/keycloakoidc.(*KeyCloakClient).getFromKeyCloakByID(0xc00450f6a0?, {0xc00589ef54, 0x24}, {0xc00589ef4d, 0x4}, 0xc005826000) │
│     /go/src/github.com/rancher/rancher/pkg/auth/providers/keycloakoidc/keycloak_client.go:135 +0xb9                                                                      │
│ github.com/rancher/rancher/pkg/auth/providers/keycloakoidc.(*keyCloakOIDCProvider).GetPrincipal(_, {_, _}, {{{0xc00450f64a, 0x5}, {0xc00450a7b0, 0x17}}, {{0xc00450f690, │
│     /go/src/github.com/rancher/rancher/pkg/auth/providers/keycloakoidc/keycloak_provider.go:150 +0x2a5                                                                   │
│ github.com/rancher/rancher/pkg/auth/providers.GetPrincipal({_, _}, {{{0xc00450f64a, 0x5}, {0xc00450a7b0, 0x17}}, {{0xc00450f690, 0xb}, {0xc00450f650, 0x6}, ...}, ...})  │
│     /go/src/github.com/rancher/rancher/pkg/auth/providers/providers.go:164 +0x105                                                                                        │
│ github.com/rancher/rancher/pkg/auth/providerrefresh.(*refresher).refreshAttributes(0xc00a955f00, 0xc00071ab00?)                                                          │
│     /go/src/github.com/rancher/rancher/pkg/auth/providerrefresh/refresher.go:281 +0xfc8                                                                                  │
│ github.com/rancher/rancher/pkg/auth/providerrefresh.RefreshAttributes(0xc00071ab00)                                                                                      │
│     /go/src/github.com/rancher/rancher/pkg/auth/providerrefresh/daemon.go:83 +0x99                                                                                       │
│ github.com/rancher/rancher/pkg/controllers/management/auth.(*UserAttributeController).sync(0xc00cb6c680, {0xc0021641b0?, 0x0?}, 0x0?)                                    │
│     /go/src/github.com/rancher/rancher/pkg/controllers/management/auth/user_attribute_handler.go:39 +0x5f                                                                │
│ github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3.(*userAttributeController).AddHandler.func1({0xc0015cffb0?, 0xc?}, {0x3e81240?, 0xc00071ab00?})  │
│     /go/src/github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3/zz_generated_user_attribute_controller.go:155 +0x42                                  │
│ github.com/rancher/norman/controller.(*genericController).AddHandler.func1({0xc0015cffb0, 0xc}, {0x4811ef8?, 0xc00071ab00})                                              │
│     /go/pkg/mod/github.com/rancher/norman@v0.0.0-20220627222520-b74009fac3ff/controller/generic_controller.go:60 +0x18b                                                  │
│ github.com/rancher/lasso/pkg/controller.SharedControllerHandlerFunc.OnChange(0xc001a4cca0?, {0xc0015cffb0?, 0xc004ad6c48?}, {0x4811ef8?, 0xc00071ab00?})                 │
│     /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20220628160937-749b3397db38/pkg/controller/sharedcontroller.go:29 +0x38                                                  │
│ github.com/rancher/lasso/pkg/controller.(*SharedHandler).OnChange(0xc000c70280, {0xc0015cffb0, 0xc}, {0x4811ef8, 0xc00071ab00})                                          │
│     /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20220628160937-749b3397db38/pkg/controller/sharedhandler.go:75 +0x23f                                                    │
│ github.com/rancher/lasso/pkg/controller.(*controller).syncHandler(0xc0008df8c0, {0xc0015cffb0, 0xc})                                                                     │
│     /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20220628160937-749b3397db38/pkg/controller/controller.go:233 +0x93                                                       │
│ github.com/rancher/lasso/pkg/controller.(*controller).processSingleItem(0xc0008df8c0, {0x34feaa0?, 0xc007889b50?})                                                       │
│     /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20220628160937-749b3397db38/pkg/controller/controller.go:214 +0x105                                                      │
│ github.com/rancher/lasso/pkg/controller.(*controller).processNextWorkItem(0xc0008df8c0)                                                                                  │
│     /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20220628160937-749b3397db38/pkg/controller/controller.go:191 +0x46                                                       │
│ github.com/rancher/lasso/pkg/controller.(*controller).runWorker(0xc003ca56a0?)                                                                                           │
│     /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20220628160937-749b3397db38/pkg/controller/controller.go:180 +0x25                                                       │
│ k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0046f1680?)                                                                                                      │
│     /go/pkg/mod/k8s.io/apimachinery@v0.24.5/pkg/util/wait/wait.go:155 +0x3e                                                                                              │
│ k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x4807960, 0xc0049f88a0}, 0x1, 0xc001680a80)                                                                       │
│     /go/pkg/mod/k8s.io/apimachinery@v0.24.5/pkg/util/wait/wait.go:156 +0xb6                                                                                              │
│ k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)                                                                                         │
│     /go/pkg/mod/k8s.io/apimachinery@v0.24.5/pkg/util/wait/wait.go:133 +0x89                                                                                              │
│ k8s.io/apimachinery/pkg/util/wait.Until(0xc003ca57d0?, 0xc57806?, 0xc00494af60?)                                                                                         │
│     /go/pkg/mod/k8s.io/apimachinery@v0.24.5/pkg/util/wait/wait.go:90 +0x25                                                                                               │
│ created by github.com/rancher/lasso/pkg/controller.(*controller).run                                                                                                     │
│     /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20220628160937-749b3397db38/pkg/controller/controller.go:148 +0x2a7                                                      │
│ panic: runtime error: index out of range [1] with length 1 [recovered]                                                                                                   │
│     panic: runtime error: index out of range [1] with length 1 
```

**Comment by jaredkipe** [comment]
Duplicate of https://github.com/rancher/rancher/issues/38683 which seems solved, so can I expect this in 2.7.2 and 2.6.1x?

**Comment by samjustus** [comment]
yes
